hugegraph-api/src/main/java/org/apache/hugegraph/core/GraphManager.java

```diff
@@ -428,13 +443,43 @@ private void serverStarted(HugeConfig config) {
                         "The server role can't be null or empty");
         this.server = IdGenerator.of(server);
         this.role = NodeRole.valueOf(role.toUpperCase());
+
+        initRoleStateMachine(config, server);
+
         for (String graph : this.graphs()) {
             HugeGraph hugegraph = this.graph(graph);
             assert hugegraph != null;
             hugegraph.serverStarted(this.server, this.role);
         }
     }
 
+    private void initRoleStateMachine(HugeConfig config, String server) {
+        try {
+            if (!(this.authenticator() instanceof StandardAuthenticator)) {
+                LOG.info("Current not support role state machine");
+                return;
+            }
+        } catch (IllegalStateException e) {
+            LOG.info("Current not support role state machine");
+            return;
+        }
+
+        E.checkArgument(this.roleStateMachine == null, "Repetition init");
+        Config roleStateMachineConfig = new HugeRoleStateMachineConfig(server,
+                                            config.get(ServerOptions.EXCEEDS_FAIL_COUNT),
+                                            config.get(ServerOptions.RANDOM_TIMEOUT_MILLISECOND),
+                                            config.get(ServerOptions.HEARTBEAT_INTERVAL_SECOUND),
+                                            config.get(ServerOptions.EXCEEDS_WORKER_COUNT),
+                                            config.get(ServerOptions.BASE_TIMEOUT_MILLISECOND));
+        StandardHugeGraph graph = (StandardHugeGraph) this.authenticator().graph().hugegraph();
+        RoleTypeDataAdapter adapter = new RoleTypeDataAdapterImpl(graph.hugeGraphParams());
+        this.roleStateMachine = new RoleElectionStateMachineImpl(roleStateMachineConfig, adapter);
```

Comment:
this.roleStateMachine = graph.roleStateMachine()

Reply:
there are multiple graphs, but only one state machine